### PR TITLE
TOS-96-Add Popular Products section

### PR DESCRIPTION
**Pull request by KarlAnna**
## Ticket
Link to ticket: https://teamchallengeonlinestore.atlassian.net/browse/TOS-96

## Description
Added Popular Products section and components for product list and product card

- Add PopularProducts component
- Add ProductsList component
- Add Card component
- Change onClick prop interface for Button and IconBtn

## Screenshots

https://www.loom.com/share/b40850e89db64257b0557d3bd4f10ab8?sid=d168914b-d3d2-44c9-b5d7-396ec6c30de0

## Checklist
- [X] Did you test it?
- [X] Did you add a screenshot?
- [X] Did you fix the merge conflict?
- [X] Did you fill the PR template?
- [x] Did you review this PR on your own?

#### Last step if the ABOVE STEPS are COMPLETED!
- [x] Did you share the link in a group chat?

### CODE REVIEW

File: app/components/ProductsList/ProductsList.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+interface ProductsListProps {
+    products: Product[];
+    infiniteScroll?: boolean;
+}
+
+const ProductsList = ({ products, infiniteScroll = false }: ProductsListProps) => {
+
+    return (
+        <div className={styles.container}>
+            <ul className={styles.cardList}>
```
Comment (dev1B) [inline app/components/ProductsList/ProductsList.tsx:23]: Do you really need this list?

File: app/components/Card/Card.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+                <Image src={img} alt={title} fill />
+            </div>
+            <div className={styles.container}>
+                <h4 className={styles.title}>{title}</h4>
+                <IconBtn onClick={handleFavorite} Icon={isFavorite ? RiHeartFill : RiHeartLine} iconClassName={cn(styles.heart, { [styles.activeHeart]: isFavorite })} type='rounded' />
+            </div>
+            <div className={styles.container}>
+                <div>
+                    {stars.map((filled, index) =>
+                        filled ? <RiStarSFill key={index} className={styles.filledStar} /> : <RiStarSLine key={index} className={styles.star} />
```
Comment (dev1B) [inline app/components/Card/Card.tsx:49]: Do you expect 3.3, 4.2 (float value) rating?

File: app/components/Card/Card.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}
+
+const Card = ({
+    img, title, rating, reviews, price
+}: CardProps) => {
+    const [isFavorite, setIsFavorite] = useState(false)
+    const formattedPrice = price.toFixed(2);
+
+    const handleFavorite = (e: React.MouseEvent<HTMLButtonElement>) => {
+        e.preventDefault();
```
Comment (dev1B) [inline app/components/Card/Card.tsx:27]: Why did you add this?

File: app/components/Card/Card.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        setIsFavorite(!isFavorite);
+    }
+
+    const handleCart = (e: React.MouseEvent<HTMLButtonElement>) => {
+        e.preventDefault();
```
Comment (dev1B) [inline app/components/Card/Card.tsx:32]: Why did you add `e.preventDefault()`?

**Review by dev1B — CHANGES_REQUESTED** [review]

File: app/components/Card/Card.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    const stars = Array.from({ length: 5 }, (_, index) => index < rating);
+
+    return (
+        <Link href='/' className={styles.card}>
+            <div className={styles.imgContainer}>
+                <Image src={img} alt={title} fill />
+            </div>
+            <div className={styles.container}>
+                <h4 className={styles.title}>{title}</h4>
+                <IconBtn onClick={handleFavorite} Icon={isFavorite ? RiHeartFill : RiHeartLine} iconClassName={cn(styles.heart, { [styles.activeHeart]: isFavorite })} type='rounded' />
```
Comment (darktend) [inline app/components/Card/Card.tsx:44]: maybe it's better ({cn(styles.heart, { [styles.activeHeart]: isFavorite })})}) to put it behind the return?

File: app/components/ProductsList/ProductsList.tsx (hunk 1 above)
Comment (KarlAnna) [reply app/components/ProductsList/ProductsList.tsx:23]: Do you mean component or tag <ul>?
If it's a tag, why shouldn't it be here? It's a list of products

File: app/components/Card/Card.tsx (hunk 5 above)
Comment (dev1B) [reply app/components/Card/Card.tsx:44]: You can use it as const 

const favoriteIcon = useMemo( () => 
<IconBtn onClick={handleFavorite} Icon={isFavorite ? RiHeartFill : RiHeartLine} iconClassName={cn(styles.heart, { [styles.activeHeart]: isFavorite })} type='rounded' />

, [deps])

**Pull request merged**